backend/onedrive/onedrive.go

```diff
@@ -432,14 +480,18 @@ func (f *Fs) readMetaDataForPathRelativeToID(ctx context.Context, normalizedID s
 // readMetaDataForPath reads the metadata from the path (relative to the absolute root)
 func (f *Fs) readMetaDataForPath(ctx context.Context, path string) (info *api.Item, resp *http.Response, err error) {
 	firstSlashIndex := strings.IndexRune(path, '/')
-
-	if f.driveType != driveTypePersonal || firstSlashIndex == -1 {
+	if (f.driveType != driveTypePersonal && f.opt.Region != "cn") || firstSlashIndex == -1 {
```

Comment:
```suggestion
	if f.driveType != driveTypePersonal || firstSlashIndex == -1 {
```

As the difference between the Chinese version and the global is only `:/` and `/children/`, we can still catch the Chinese case in this block. The remaining code outside this block is intended for OneDrive Personal only, which has "shared with me" folders. They don't exist in OneDrive for Business.

Reply:
Here `/children/` in the Chinese version won't work when called `/items/<id>/children/<dir>/<file>`.It only allows one level of subdirectory.
 So I think maybe the solution for OneDrive personal  "shared with me" folders is more suitable.

Reply:
I don't think it's assumed in the code that `readMetaDataForPathRelativeToID` is called with only one level of subdirectory. The special handling for shared folders in `readMetaDataForPath` only makes sure that if we're querying an item inside a folder, we guarantee that it's queried with a path relative to the ID of *at least* the first level directory in its path. That is to say, if you call `readMetaDataForPath(ctx, "a/b/c")`, it's possible that the function in turn calls `readMetaDataForPathRelativeToID(ctx, idOfA, "b/c")`.

Reply:
Then maybe a brand new implementation is needed. I'll try to work for it.

Reply:
Now there are to solutions. The first is to implement a function to query subdirectories one by one, and another is to switch to `/me/drive` in the Chinese version.
I'll try the first one first.